system/Commands/Utilities/Routes/FilterFinder.php

```diff
@@ -50,6 +51,12 @@ public function find(string $uri): array
     {
         $this->filters->reset();
 
+        $isLocaleExist = strpos($uri, '{locale}') !== false;
```

Comment:
This change is not acceptable.
Because the param `$uri` is "URI path to find filters for".
`{locale}/admin/settings` is not a URI path.
The URI path is like `ja/admin/settings` or `fr/admin/settings`.

This change will break the `filter:check` command.

Reply:
See debug info.
```php
// Routes.php
/**
 * @var RouteCollection $routes
 */
$routes->useSupportedLocalesOnly(true);
$routes->get('/{locale}/home', 'Home::index');

// FilterFinder.php
    public function find(string $uri): array
    {
        d($uri);
        $this->filters->reset();

        // Fix for the search filters command
        $isSupportedLocaleOnly = false;
```
Output:
![image](https://github.com/codeigniter4/CodeIgniter4/assets/17183530/f0913c61-ddb1-467a-a0e8-781da1631575)

Set **ja, fr** crash my test.
```diff
1) CodeIgniter\Commands\Utilities\Routes\FilterFinderTest::testFindFiltersWithSupportedLocalesOnly
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
 Array &0 (
     'before' => Array &1 (
-        0 => '!csrf'
+        0 => 'csrf'
     )
     'after' => Array &2 (
-        0 => '!toolbar'
+        0 => 'toolbar'
     )
 )
```

Reply:
And filter:checl normal output.
**useSupportedLocalesOnly = true**
```console
$ ./spark filter:check get en/home

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-11-06 10:41:42 UTC+00:00

+--------+---------+----------------+---------------+
| Method | Route   | Before Filters | After Filters |
+--------+---------+----------------+---------------+
| GET    | en/home |                | toolbar       |
+--------+---------+----------------+---------------+

$ ./spark filter:check get ja/home

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-11-06 10:43:31 UTC+00:00

Can't find a route: "GET ja/home"
```

**useSupportedLocalesOnly = false**

```console
$ ./spark filter:check get en/home

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-11-06 10:42:01 UTC+00:00

+--------+---------+----------------+---------------+
| Method | Route   | Before Filters | After Filters |
+--------+---------+----------------+---------------+
| GET    | en/home |                | toolbar       |
+--------+---------+----------------+---------------+

$ ./spark filter:check get ja/home

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-11-06 10:42:03 UTC+00:00

+--------+---------+----------------+---------------+
| Method | Route   | Before Filters | After Filters |
+--------+---------+----------------+---------------+
| GET    | ja/home |                | toolbar       |
+--------+---------+----------------+---------------+

aleksandr@aleksandr-debian:~/www/codeigniter$ 
```

Reply:
No. FilterFinder's responsibility is to find filters for a URI path.
If you pass `{locale}/home` to it, it is a URI path of e.g., `http://localhost/{locale}/home`. 
It does not mean `ja/home` or `fr/home`.

Don't change the behavior.

Reply:
I don't understand what the problem is... Is there a specific example with a behavior error?
Now everything is working correctly: with and without the `useSupportedLocalesOnly` option.

Reply:
We test route as `useSupportedLocalesOnly = true/false`:
```console
$ ./spark filter:check get {locale}/home

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-11-06 11:03:00 UTC+00:00

+--------+---------------+----------------+---------------+
| Method | Route         | Before Filters | After Filters |
+--------+---------------+----------------+---------------+
| GET    | {locale}/home |                | !toolbar      |
+--------+---------------+----------------+---------------+

$ ./spark filter:check get {locale}/home

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-11-06 11:03:22 UTC+00:00

+--------+---------------+----------------+---------------+
| Method | Route         | Before Filters | After Filters |
+--------+---------------+----------------+---------------+
| GET    | {locale}/home |                | toolbar       |
+--------+---------------+----------------+---------------+
```

Reply:
`filter:check` should not output `!`. It must always return filters to apply. 
```console
$ php spark filter:check get {locale}/home

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-11-06 11:10:54 UTC+00:00

+--------+---------------+----------------+---------------+
| Method | Route         | Before Filters | After Filters |
+--------+---------------+----------------+---------------+
| GET    | {locale}/home | !invalidchars  | !toolbar      |
+--------+---------------+----------------+---------------+
```

Reply:
Do not change the meaning of `$uri`. It is the literal URI path, no meta characters or placeholders.

Reply:
So this is what we wanted, isn't it? Show POSSIBLE filters for a GROUP of languages. If you check the exact language everything will be fine

Reply:
It is for `routes` command. Not for `filter:check`.
`routes` will show inaccurate filter info, but useful.
But `filter:check` should always show accurate filter info.